File: frontend/src/types/vlog.ts
export type ProcessingStatus = 'pending' | 'transcribing' | 'planning' | 'ready' | 'failed'

export interface Vlog {
  id: string
  platform: 'youtube' | 'instagram'
  platform_video_id: string
  title: string
  description: string | null
  thumbnail_url: string | null
  video_url: string | null
  channel_name: string | null
  duration_seconds: number | null
  published_at: string | null
  view_count: number | null
  like_count: number | null
  destinations: string[]
  travel_styles: string[]
  processing_status: ProcessingStatus
  created_at: string
  itinerary_id: string | null
}

export interface FeedPage {
  vlogs: Vlog[]
  next_cursor: string | null
  total: number
}

export interface VlogInteraction {
  vlog_id: string
  action: 'view' | 'like' | 'save' | 'share' | 'book_started'
}

File: backend/app/schemas/vlog.py
from pydantic import BaseModel
from typing import Optional, List
from datetime import datetime
from enum import Enum


class ProcessingStatus(str, Enum):
    pending = "pending"
    transcribing = "transcribing"
    planning = "planning"
    ready = "ready"
    failed = "failed"


class VlogResponse(BaseModel):
    id: str
    platform: str
    platform_video_id: str
    title: str
    description: Optional[str] = None
    thumbnail_url: Optional[str] = None
    video_url: Optional[str] = None
    channel_name: Optional[str] = None
    duration_seconds: Optional[int] = None
    published_at: Optional[datetime] = None
    view_count: Optional[int] = None
    like_count: Optional[int] = None
    destinations: List[str] = []
    travel_styles: List[str] = []
    processing_status: ProcessingStatus
    itinerary_id: Optional[str] = None
    created_at: datetime


class FeedPage(BaseModel):
    vlogs: List[VlogResponse]
    next_cursor: Optional[str] = None
    total: int


class VlogInteractionRequest(BaseModel):
    vlog_id: str
    action: str
    duration_watched_seconds: Optional[int] = None
